### DISCUSSION TITLE
How to reuse laravel's base throughout many applications?

### DISCUSSION THREAD

**Discussion by Nefcanto** [laravel/framework#37671] [Q&A]
I want to create these applications using laravel:

- Online shop for handcrafted products
- Online consulting for cryptocurrency
- Paid education and tutoring
- ...

Let's say I have 20 apps in mind to create.

All of these apps have these common features:

- They all use the same authentication technology
- They all have the same exception handling strategies
- They all use the same set of middleware
- They all have the same HTTP and Console kernels
- They all have the same providers
- They all have the same config in
    - auth
    - cors
    - hashing
    - logging
    - sessions
    - ...
- They all have common dependencies in their `composer.json`

As you can see, while my applications are similar, I have to repeat myself a lot.

What I want is to centralize all of the common aspects **ACROSS** many applications.

In simple English terms, I want to have a laravel project that is called `HandyShop` and when I open it in VS Code, I don't want to see `cors.php` and `middleware folder and ...

I want to tell laravel **hey laravel, please read all of these common features from this base package**.

I have managed to centralize logic using packages. For example, I have **payment** package, **ticketing** package, etc. But I also want to centralize infrastructure.

Please guide me on how can I start? How can I do this?

**Comment by s-patompong** [comment]
IMHO, I think you're heading the right way using packages to reuse the business logic. You can also implement those 20 apps using one Laravel repo though, create one API routes and multiple other routes for each subdomain (which reuse Kernal, API routes, etc.) if that makes sense.

**Comment by Nefcanto** [comment]
@s-patompong , thank you for the suggestion. Unfortunately it's not possible to have all of them being in one git repo. Based on some organizational policies, we need to use different git repos. But those git repos CAN use a central infra repo.

**Comment by denisdulici** [comment]
You can use a modular structure for that purpose. We've applied in [Akaunting](https://github.com/akaunting/akaunting) via our [laravel-module](https://github.com/akaunting/laravel-module) package and it works like a charm.
